content/ExceptionHandling/ExceptionHandling.q.md

```diff
@@ -0,0 +1,1395 @@
+
+In the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block on lines 13, 14, and 15, what is `catch`?
+
+ - It is a class.
+
+ - It is a keyword.
+
+ - It is a function.
+
+ - It is an operator.
+
+ - It is a parameter.
+
+
+/// type=SS, id=f799b3df-bc73-44cd-9fa5-92edc0d7700e, answer=[1]
+
+In the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block on lines 13, 14, and 15, what is `Exception`?
+
+ - It is a class.
+
+ - It is a keyword.
+
+ - It is a function.
+
+ - It is an operator.
+
+ - It is a parameter.
+
+
+/// type=SS, id=703bc42f-1bd3-4799-839e-934933a68ccd, answer=[5]
+
+In the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block on lines 13, 14, and 15, what is `$e`?
+
+ - It is a class.
+
+ - It is a method.
+
+ - It is a keyword.
+
+ - It is a parameter.
+
+ - It is an instance of the `Exception` class.
+
+
+/// type=SS, id=3f7fb2fa-faf4-4348-b025-14687a3432a6, answer=[5]
+
+In the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block on lines 13, 14, and 15, what does `Exception $e` do?
+
+ - It displays the value of the `$e` object.
+
+ - It sets the value of the `Exception` class to `$e`.
+
+ - It returns the `$e` object of the `Exception` class.
+
+ - It accesses the value of `$e` in the `Exception` class.
+
+ - It creates the `$e` object as an instance of the `Exception` class.
+
+
+/// type=SS, id=43361ac8-3f6c-4dc7-b20e-231f16b22896, answer=[4]
+
+In the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block on lines 13, 14, and 15, what does `$e->getMessage()` do?
+
+ - It returns the `getMessage()` method of the `$e` object.
+
+ - It creates the `getMessage()` method of the `$e` object.
+
+ - It displays the `getMessage()` method of the `$e` object.
+
+ - It accesses the `getMessage()` method of the `$e` object.
+
+ - It removes the `getMessage()` method of the `$e` object.
+
+
+/// type=MS, id=f7105568-7eaf-4926-89ee-c8095c3363dc, answer=[1,4]
+
+On lines 13, 14, and 15, what does the `catch (Exception $e) { echo "Exception caught: " . $e->getMessage(); }` block do?
+
+ - It catches the possible exception that occurs in the `try` block.
+
+ - It throws an exception when an error occurs in `Exception $e`.
+
+ - It returns an exception when an error occurs in `Exception $e`.
+
+ - It executes the `echo` statement if an exception occurs in the `try` block.
+
+ - It enables the `divide()` function to throw an exception if an error occurs.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+function divide($dividend, $divisor)
+{
+    if (!is_numeric($divisor) || !is_numeric($dividend)) {
+        throw new InvalidArgumentException("Invalid argument value provided");
+    } elseif ($divisor == 0) {
+        throw new Exception("The value '0' as a divisor is not allowed.");
+    } else {
+        return ($dividend / $divisor);
+    }
```

Comment:
We can simplify this by having only 2 `if` conditions and removing the `elseif` and `else`. since it would no longer proceed if the first condition is `true`.